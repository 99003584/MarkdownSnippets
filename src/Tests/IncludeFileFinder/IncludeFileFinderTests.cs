using System.Collections.Concurrent;
using System.IO;
using System.Linq;
using Xunit;
using Xunit.Abstractions;

public class IncludeFileFinderTests :
    XunitApprovalBase
{
    [Fact]
    public void Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "IncludeFileFinder/Nested");
        var finder = new IncludeFileFinder();
        var files = finder.FindFiles(directory);
        ObjectApprover.Verify(files, Scrubber.Scrub);
    }

    [Fact]
    public void Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "IncludeFileFinder/Simple");
        var finder = new IncludeFileFinder();
        var files = finder.FindFiles(directory);
        ObjectApprover.Verify(files, Scrubber.Scrub);
    }

    [Fact]
    public void VerifyLambdasAreCalled()
    {
        var directories = new ConcurrentBag<string>();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "IncludeFileFinder/VerifyLambdasAreCalled");
        var finder = new IncludeFileFinder(
            directoryFilter: path =>
            {
                directories.Add(path);
                return true;
            }
        );
        finder.FindFiles(targetDirectory);
        ObjectApprover.Verify(directories.OrderBy(file => file), Scrubber.Scrub);
    }

    public IncludeFileFinderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}