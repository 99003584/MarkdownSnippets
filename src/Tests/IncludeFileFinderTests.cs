using System.Collections.Concurrent;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using VerifyXunit;
using Xunit;

[UsesVerify]
public class IncludeFileFinderTests
{
    [Fact]
    public Task Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "IncludeFileFinder/Nested");
        IncludeFileFinder finder = new();
        var files = finder.FindFiles(directory);
        return Verifier.Verify(files);
    }

    [Fact]
    public Task Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "IncludeFileFinder/Simple");
        IncludeFileFinder finder = new();
        var files = finder.FindFiles(directory);
        return Verifier.Verify(files);
    }

    [Fact]
    public Task VerifyLambdasAreCalled()
    {
        ConcurrentBag<string> directories = new();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "IncludeFileFinder/VerifyLambdasAreCalled");
        IncludeFileFinder finder = new(
            shouldIncludeDirectory: path =>
            {
                directories.Add(path);
                return true;
            }
        );
        finder.FindFiles(targetDirectory);
        return Verifier.Verify(directories.OrderBy(file => file));
    }
}