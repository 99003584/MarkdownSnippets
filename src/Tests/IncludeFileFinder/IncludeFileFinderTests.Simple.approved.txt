[
  'root\\IncludeFileFinder\Simple\file1.include.md',
  'root\\IncludeFileFinder\Simple\file2.include.md'
]